src/A2A.AspNetCore/A2AJsonRpcProcessor.cs

```diff
@@ -138,7 +138,17 @@ private static T DeserializeOrThrow<T>(JsonElement jsonParamValue) where T : cla
             parms = null;
         }
 
-        return parms ?? throw new A2AException("Invalid parameters", A2AErrorCode.InvalidParams);
+        switch (parms)
+        {
+            case null:
+                throw new A2AException("Invalid parameters", A2AErrorCode.InvalidParams);
+            case MessageSendParams messageSendParams when messageSendParams.Message.Parts.Count == 0:
+                throw new A2AException("Message parts cannot be empty", A2AErrorCode.InvalidParams);
+            case TaskQueryParams taskQueryParams when taskQueryParams.HistoryLength < 0:

```

Comment:
w/o opening in the IDE I think `Message` and/or `.Parts` as well as `.HistoryLength` are all nullable... so I think null handling here might be necessary as well, no?

Reply:
`Message` is not nullable, marked as required and by assigned to a new instance by default:

https://github.com/a2aproject/a2a-dotnet/blob/2c2fc4a0b876b50338d89e2d0a58955809215b96/src/A2A/Models/MessageSendParams.cs#L17-L19

In case of deserialization, a `null` message would be rejected by the deserializer